src/blob/core.rs

```diff
@@ -243,16 +244,23 @@ where
     pub(crate) async fn write(blob: &ASRwLock<Self>, key: &K, record: Record) -> Result<()> {
         debug!("blob write");
         // Only one upgradable_read lock is allowed at a time
+        let (partially_serialized, header) = record.to_partially_serialized_and_header()?;
         let blob = blob.upgradable_read().await;
-        Self::write_locked(blob, key, record).await
+        Self::write_locked(blob, key, partially_serialized, header).await
     }
 
-    async fn write_mut(&mut self, key: &K, mut record: Record) -> Result<RecordHeader> {
+    async fn write_mut(
```

Comment:
I think there is no need to update `write_mut` signature. It is better to receive `Record` as it was before and only inside perform the conversion to `PartiallySerializedRecord`. 
The optimization with `PartiallySerializedRecord` was designed for `write` + `write_locked` path